### PR TITLE
Move xqueue to supervisor

### PR DESCRIPTION
**Pull request by antoviaque**
@jarv A first attempt at moving a service to supervisor -- can you check if I didn't forget something important? Once this is good to merge I'll do the same thing with the remaining services.

Btw this pull request also includes a series of fixes to get xqueue & ora to work properly, as I used the peer grading feature to check that xqueue was properly installed and functioned correctly.

### CODE REVIEW

File: playbooks/roles/supervisor/templates/supervisor-upstart.conf.j2 [hunk 1]
```diff
@@ -4,5 +4,4 @@ start on runlevel [2345]
 stop on runlevel [!2345]
 
 respawn
-setuid {{ common_web_user }}
```
Comment (jarv) [inline playbooks/roles/supervisor/templates/supervisor-upstart.conf.j2]: We were thinking that supervisord as well as all the processes it controls would be run as `{{ common_web_user }}`.
This allows us to connect to it as that user which allows us to manage the processes as www-data, instead of root.

File: playbooks/roles/xqueue/templates/xqueue.conf.j2 [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
-env PORT={{ xqueue_gunicorn_port }}
-env ADDRESS={{ xqueue_gunicorn_host }}
-env LANG=en_US.UTF-8
-env DJANGO_SETTINGS_MODULE=xqueue.aws_settings
-env SERVICE_VARIANT="xqueue"
 
+user={{ xqueue_user }}
+directory={{ xqueue_code_dir }}
+
+environment=PID=/var/tmp/xqueue.pid,PORT={{ xqueue_gunicorn_port }},ADDRESS={{ xqueue_gunicorn_host }},LANG=en_US.UTF-8,DJANGO_SETTINGS_MODULE=xqueue.aws_settings,SERVICE_VARIANT="xqueue"
```
Comment (jarv) [inline playbooks/roles/xqueue/templates/xqueue.conf.j2]: even though it wasn't before I think we want LANG to be a var that can be overridden, something like `{{ XSERVER_LANG }}` (we did the same for edxapp already)

File: playbooks/roles/xqueue/templates/xqueue.conf.j2 [hunk 3]
```diff
@@ -1,25 +1,16 @@
...
-respawn
-respawn limit 3 30
-
-env PID=/var/tmp/xqueue.pid
 {% if ansible_processor|length > 0 %}
-env WORKERS={{ ansible_processor|length * 2 }}
+command={{ xqueue_venv_dir }}/bin/gunicorn --preload -b {{ xqueue_gunicorn_host }}:{{ xqueue_gunicorn_port }} -w {{ ansible_processor|length * 2 }} --timeout=300 --pythonpath={{ xqueue_code_dir }} xqueue.wsgi
 {% else %}
-env WORKERS=2
+command={{ xqueue_venv_dir }}/bin/gunicorn --preload -b {{ xqueue_gunicorn_host }}:{{ xqueue_gunicorn_port }} -w 2 --timeout=300 --pythonpath={{ xqueue_code_dir }} xqueue.wsgi
```
Comment (jarv) [inline playbooks/roles/xqueue/templates/xqueue.conf.j2]: for this and the consumer you can use `{{ xqueue_venv_bin }}/gunicorn`

Comment (jarv) [inline playbooks/roles/xqueue/templates/xqueue.conf.j2]: maybe add a new var for `{{ xqueue_gunicorn_workers }}` ?

File: playbooks/roles/ora/tasks/ease.yml [hunk 4]
```diff
@@ -54,7 +54,7 @@
 
 - name: ora | install ease python package
   shell: >
-    {{ ora_ease_venv_dir }}/bin/activate; cd {{ ora_ease_code_dir }}; python setup.py install
+    . {{ ora_ease_venv_dir }}/bin/activate; cd {{ ora_ease_code_dir }}; python setup.py install
```
Comment (jarv) [inline playbooks/roles/ora/tasks/ease.yml]: Can the ora changes be moved into a different PR?
I think this is fine, `{{ ora_eas_venv_bin }}/python setup.py install` will also work I think without sourcing the activate script.

File: playbooks/roles/supervisor/templates/supervisor-upstart.conf.j2 (hunk 1 above)
Comment (antoviaque) [inline playbooks/roles/supervisor/templates/supervisor-upstart.conf.j2]: Then that means ora needs to be run as `{{ common_web_user }}` too no? Without it supervisord can't suid and gives the error:

```
supervisor: couldn't setuid to 1004: Can't drop privilege as nonroot user
supervisor: child process was not spawned
```

Comment (jarv) [inline playbooks/roles/supervisor/templates/supervisor-upstart.conf.j2]: that's correct and what I am currently doing for other services (forum, lms, cms) that are running under supervisord.

File: playbooks/roles/ora/tasks/ease.yml (hunk 4 above)
Comment (antoviaque) [inline playbooks/roles/ora/tasks/ease.yml]: Sure - I've put the relevant commits in this pull request: https://github.com/edx/configuration/pull/382

**Comment by antoviaque** [comment]
@jarv Made the changes, let me know if you see anything else/left.

File: playbooks/roles/xqueue/templates/xqueue.conf.j2 [hunk 5]
```diff
@@ -1,25 +1,16 @@
...
-respawn
-respawn limit 3 30
-
-env PID=/var/tmp/xqueue.pid
 {% if ansible_processor|length > 0 %}
-env WORKERS={{ ansible_processor|length * 2 }}
+command={{ xqueue_venv_bin }}/gunicorn --preload -b {{ xqueue_gunicorn_host }}:{{ xqueue_gunicorn_port }} -w {{ ansible_processor|length * 2 }} --timeout=300 --pythonpath={{ xqueue_code_dir }} xqueue.wsgi
 {% else %}
-env WORKERS=2
+command={{ xqueue_venv_bin }}/gunicorn --preload -b {{ xqueue_gunicorn_host }}:{{ xqueue_gunicorn_port }} -w 2 --timeout=300 --pythonpath={{ xqueue_code_dir }} xqueue.wsgi
```
Comment (antoviaque) [inline playbooks/roles/xqueue/templates/xqueue.conf.j2:6]: For your remark here, about adding a variable `xqueue_gunicorn_workers` instead of the test, I agree it would be better. But I'm not sure how it would be possible to declare this in `defaults/main.yml`, since we need to use a conditionals (`{% if ansible_processor|length > 0 %}`)?

**Comment by jarv** [comment]
lgtm
